detections/gcp_detect_oauth_token_abuse.yml

```diff
@@ -0,0 +1,22 @@
+author: "Rod Soto, Splunk"
+date: "2020-09-01"
+description: "This search provides detection of possible GCP Oauth token abuse. GCP Oauth token without time limit can be exfiltrated and reused for keeping access sessions alive without further control of authentication, allowing attackers to access and move laterally."
+how_to_implement: "You must install splunk GCP add-on. This search works with gcp:pubsub:message logs"
+id: a7e9f7bb-8901-4ad0-8d88-0a4ab07b1972
+known_false_positives: "GCP Oauth token abuse detection will only work if there are access policies in place along with audit logs."
+name: "gcp detect oauth token abuse"
+references:
+  - https://www.netskope.com/blog/gcp-oauth-token-hijacking-in-google-cloud-part-1
+  - https://www.netskope.com/blog/gcp-oauth-token-hijacking-in-google-cloud-part-2
+search: '`sourcetype=google:gcp:pubsub:message type.googleapis.com/google.cloud.audit.AuditLog |table protoPayload.@type protoPayload.status.details{}.@type protoPayload.status.details{}.violations{}.callerIp protoPayload.status.details{}.violations{}.type protoPayload.status.message  | `gcp_detect_oauth_token_abuse`'
```

Comment:
the last macro needs "_filter"
the sourcetype in the beginning should be a macro